src/main/java/org/ow2/proactive/connector/iaas/ShutdownHandler.java

```diff
@@ -43,18 +43,13 @@
 
     @PreDestroy
     public synchronized void removeAllInfrastructures() {
-        infrastructureService.getAllSupportedInfrastructure()
-                             .values()
-                             .stream()
-                             .filter(infrastructure -> infrastructure.isToBeRemovedOnShutdown())
-                             .forEach(infrastructure -> {
-                                 try {
-                                     infrastructureService.deleteInfrastructure(infrastructure);
-                                 } catch (Exception e) {
-                                     logger.error("Shutdown ERROR when trying to delete infrastructure : " +
-                                                  infrastructure, e);
-                                 }
-                             });
+        infrastructureService.getAllSupportedInfrastructure().values().forEach(infrastructure -> {
+            try {
+                infrastructureService.deleteInfrastructure(infrastructure, infrastructure.isToBeRemovedOnShutdown());
```

Comment:
done in f42bf6d3264c4bb36d20d78a6a63c60f98d60d55
The connector-iaas data are not persistent and thus caches are lost when shutted down, but anyway I think its more clean to still use a regular method `deleteInfrastructure` to deregister the infrastructure and then if the boolean is set it destroys the created instances by calling `deleteInfrastructureWithCreatedInstances`